lib/active_model/serializer/associations.rb

```diff
@@ -74,7 +74,8 @@ def has_one(name, options = {}, &block) # rubocop:disable Style/PredicateName
         # @api private
         #
         def associate(reflection)
-          self._reflections << reflection
+          key = reflection.options[:key]
+          key ? self._reflections[key] = reflection : self._reflections[reflection.name] = reflection
```

Comment:
this may be something that AMS needs to change in the future then. Cause this is silly. haha

key should be :comments for the first
and key should be last_comments for the second.